File: CS/Q248298/Form1.cs
using System;
using Northwind;
using DevExpress.XtraEditors;
using DevExpress.XtraGrid.Views.Base;

namespace Q248298 {
    public partial class MainForm : XtraForm {
        public MainForm() {
            InitializeComponent();
        }

        private void OnMainViewFocusedRowChanged(object sender, FocusedRowChangedEventArgs e) {
            Categories category = ((ColumnView)sender).GetFocusedRow() as Categories;
            detailGrid.DataSource = category == null ? null : category.Products;
        }
    }
}

File: VB/Q248298/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports Northwind
Imports DevExpress.XtraEditors
Imports DevExpress.XtraGrid.Views.Base

Namespace Q248298
	Partial Public Class MainForm
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub OnMainViewFocusedRowChanged(ByVal sender As Object, ByVal e As FocusedRowChangedEventArgs) Handles mainView.FocusedRowChanged
			Dim category As Categories = TryCast((CType(sender, ColumnView)).GetFocusedRow(), Categories)
			If category Is Nothing Then
				detailGrid.DataSource = Nothing
			Else
				detailGrid.DataSource = category.Products
			End If
		End Sub
	End Class
End Namespace